### ISSUE TITLE
raft_start(): io: closed segment xxx is past last snapshot xxx

### ISSUE THREAD

**Issue by MathieuBordere** [canonical/raft#431]
Something weird on the LXD forums https://discuss.linuxcontainers.org/t/how-to-recover-from-failed-lxd-snap-auto-update/17351/3

We've seen those kinds of issues before, but this time, it looks like something else, these are the contents of the data directory:

```
Jun 08 2023 12:34:16 root@pogo:/var/snap/lxd/common/lxd/database/global# ll
drwxr-x--- 4 root root    4096 Jun  7 22:48 ./
drwx------ 4 root root    4096 Jun  7 22:49 ../
-rw------- 1 root root 8377384 Feb 22 00:33 0000000000014337-0000000000015151
-rw------- 1 root root 1033208 Feb 22 23:17 0000000000015152-0000000000015251
-rw------- 1 root root 1169264 Feb 23 22:19 0000000000015252-0000000000015360
-rw------- 1 root root 8381408 Mar  3 17:19 0000000000015361-0000000000016173
-rw------- 1 root root 2182112 Mar  5 18:19 0000000000016174-0000000000016384
-rw------- 1 root root 8377344 Mar 13 14:19 0000000000016385-0000000000017198
-rw------- 1 root root 2149152 Mar 15 14:19 0000000000017199-0000000000017408
-rw------- 1 root root 5938944 Mar 21 01:33 0000000000017409-0000000000017985
-rw------- 1 root root 4587576 Mar 25 05:33 0000000000017986-0000000000018432
-rw------- 1 root root 8385552 Apr  2 01:33 0000000000018433-0000000000019246
-rw------- 1 root root 2149152 Apr  4 01:33 0000000000019247-0000000000019456
-rw------- 1 root root 8381560 Apr 11 21:33 0000000000019457-0000000000020272
-rw------- 1 root root 2149064 Apr 13 20:33 0000000000020273-0000000000020480
-rw------- 1 root root 5290168 Apr 18 19:13 0000000000020481-0000000000020996
-rw------- 1 root root 5228120 Apr 23 14:13 0000000000020997-0000000000021504
-rw------- 1 root root 8381448 May  1 09:14 0000000000021505-0000000000022318
-rw------- 1 root root 2165624 May  3 09:14 0000000000022319-0000000000022528
-rw------- 1 root root 8380760 May 10 23:14 0000000000022529-0000000000023332
-rw------- 1 root root 421616 Jun  7 21:40 0000000000023333-0000000000023374
-rw------- 1 root root   4184 Jun  7 21:40 0000000000023375-0000000000023375
-rw------- 1 root root 226952 Jun  7 21:40 0000000000023376-0000000000023392
-rw------- 1 root root 408608 Jun  7 21:40 0000000000023376-0000000000023407
-rw------- 1 root root  651264 May 31 12:59 db.bin
-rw------- 1 root root      32 Oct  3  2022 metadata1
-rw------- 1 root root 8388608 Jun  8 11:49 open-1
-rw------- 1 root root 8388608 Jun  7 22:48 open-2
-rw------- 1 root root 8388608 Jun  7 22:48 open-3
-rw------- 1 root root   72150 Apr 23 14:13 snapshot-1-21504-8809030973
-rw------- 1 root root      56 Apr 23 14:13 snapshot-1-21504-8809030973.meta
-rw------- 1 root root  107155 May  3 09:14 snapshot-1-22528-9655034584
-rw------- 1 root root      56 May  3 09:14 snapshot-1-22528-9655034584.meta
```

Notice that there are **2** segments with start index  `0000000000023376`.

The theory is that it could have been introduced by a recent change as the LXD users are on the `edge` channel and the LXD folks haven't seen these issues in 1.5 years.

CC @tomponline

**Comment by MathieuBordere** [comment]
possibly introduced on or before May 31st 

https://github.com/lxc/lxd-pkg-snap/commit/7b9f47091ce5e99fc034075442c0217deb79a0f0

**Comment by MathieuBordere** [comment]
```
rw------- 1 root root 421616 Jun  7 21:40 0000000000023333-0000000000023374
-rw------- 1 root root   4184 Jun  7 21:40 0000000000023375-0000000000023375
-rw------- 1 root root 226952 Jun  7 21:40 0000000000023376-0000000000023392
-rw------- 1 root root 408608 Jun  7 21:40 0000000000023376-0000000000023407
```

The user reports that he has reconstructed the contents of the data folder, so "date/times are not the actual".

But based on this, it looks like there was an ungraceful shutdown which left an (or multiple) open segment)s) (otherwise we wouldn't see the segment with a single entry). So maybe there is something going on with the closing of multiple open segments at startup, but I'm just guessing here.

**Comment by MathieuBordere** [comment]
possibly fixed by #435, reopen if it pops up again.
